astropy/io/fits/header.py

```diff
@@ -221,21 +221,11 @@ def __delitem__(self, key):
                 indices = self._rvkc_indices
 
             if key not in indices:
-                if _is_astropy_internal():
```

Comment:
Oh, and another thing I meant to point out....

Now that this is removed, the `_is_astropy_internal` function isn't used anywhere, so you can just remove this function entirely too (as well as the imports of `inspect` and `sys` since they aren't used anywhere in this module once `_is_astropy_internal` is removed.